### PR TITLE
D2K - Fix Tiles errors in Atreides Mission 2s and add production/repair for their AIs.

### PR DESCRIPTION
**Pull request by MustaphaTR**
~Depends on #12749 because of the tile changes.~ I reimported the maps added with new tiles at that PR.

Edit: Also about AI, AI wasn't able to build in original game. But i think it was a bug of original game instead of an intented thing as it is only the case for Atreides Mission 2s, but not Harkonnen or Ordos.

### CODE REVIEW

**Comment by reaperrr** [comment]
#12749 was merged.

**Comment by ltem** [comment]
AI is working but could you recheck the right border of Atreides 2a? I think I found a tile error there.

OpenRA:
image
GruntsMod:
image

**Comment by MustaphaTR** [comment]
Updated.

**Review by ltem — APPROVED** [review]

**Comment by ltem** [comment]
:+1:

File: mods/d2k/maps/atreides-02a/atreides02a-AI.lua [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+			SendAttack()
+		end
+	end)
+end
+
+ActivateAI = function()
+	Trigger.AfterDelay(0, InitAIUnits)
+
+	-- Finish the upgrades first before trying to build something
+	Trigger.AfterDelay(DateTime.Seconds(14), function()
```
Comment (abcdefg30) [inline mods/d2k/maps/atreides-02a/atreides02a-AI.lua]: The AI isn't building any upgrades in this mission, so this can go.

Comment (abcdefg30) [inline mods/d2k/maps/atreides-02a/atreides02a-AI.lua]: Unfortunately removing the delay introduced another regression (they don't build infantry). Seems like there was indeed a reason why I put the call to `InitAIUnits()` above in an `AfterDelay(0)`.

**Comment by abcdefg30** [comment]
I just put the entire call to `ActivateAI()` into an frame end task, that fixes the regression.

**Review by abcdefg30 — APPROVED** [review]
Looks good to me now.

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/29e502c4c23df06d9ebb714ca8ba148e9a274763...443e94754be95514ff079e8d06031cab708f9660)

**Pull request merged**